plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -295,15 +297,76 @@ def drop_dataset(self, database, schema):
         dataset = self.dataset(database, schema, conn)
         client = conn.handle
 
-        with self.exception_handler('drop dataset'):
-            client.delete_dataset(
-                dataset, delete_contents=True, not_found_ok=True
-            )
+        def _drop_tables_then_dataset():
+            for table in client.list_tables(dataset):
+                client.delete_table(table.reference)
+            client.delete_dataset(dataset)
+
+
+        self._retry_and_handle(
+          msg='drop dataset', conn=conn, fn=_drop_tables_then_dataset)
 
     def create_dataset(self, database, schema):
         conn = self.get_thread_connection()
         client = conn.handle
         dataset = self.dataset(database, schema, conn)
 
-        with self.exception_handler('create dataset'):
-            client.create_dataset(dataset, exists_ok=True)
+        # Emulate 'create schema if not exists ...'
+        try:
```

Comment:
same here - let's use `client.create_dataset(dataset, exists_ok=True)` - check the existing implementation